test/api/backoff_test.dart

```diff
@@ -51,4 +51,17 @@ void main() {
       check(maxFromAllTrials).isGreaterThan(expectedMax * 0.75);
     }
   });
+
+  test('BackoffMachine resets duration', () async {
+    final backoffMachine = BackoffMachine();
+    await measureWait(backoffMachine.wait());
+    final duration2 = await measureWait(backoffMachine.wait());
+    check(backoffMachine.waitsCompleted).equals(2);
+
+    backoffMachine.reset();
+    check(backoffMachine.waitsCompleted).equals(0);
+    final durationReset1 = await measureWait(backoffMachine.wait());
+    check(durationReset1).isLessThan(duration2);
```

Comment:
This test will flake — it'll fail about 25% of the time. For example:
```
$ time flutter test test/api/backoff_test.dart 
00:01 +1 -1: BackoffMachine.reset [E]                                            
  Expected: a Duration that:
    is less than <0:00:00.023591>
  Actual: <0:00:00.049613>
  Which: is not less than <0:00:00.023591>
  package:checks/src/checks.dart 85:9             check.<fn>
  package:checks/src/checks.dart 708:12           _TestContext.expect
  package:checks/src/extensions/core.dart 167:13  ComparableChecks.isLessThan
  test/api/backoff_test.dart 64:27                main.<fn>
```

The trouble is that the behavior of `wait` is randomized, and the range of possible durations for the second wait (or even the Nth wait for large N) overlaps with the range for the first wait.

That randomization is why the existing test for that method, above, runs 100 trials and then makes a fairly loose check on the results; the parameters are chosen, using some math that's in the comments, to ensure the flake rate is below one in a billion.

(I picked that threshold so that even in a large codebase, even with lots of randomized test cases using that threshold, the total flake rate due to randomization would be negligible compared to other sources of flakiness like races, other bugs, and infra issues.)

We totally could write a working test for this `reset` method by using the same strategy. But maybe the complexity of that is a good reason to skip the `reset` method and go with the other approach you mentioned at https://github.com/zulip/zulip-flutter/pull/564#issuecomment-2021780297 , of just resetting a `BackoffMachine?` local to null.